app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java

```diff
@@ -0,0 +1,66 @@
+package ch.epfl.sweng.erpa.activities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import com.annimon.stream.Optional;
+
+import java.io.Serializable;
+
+import javax.inject.Inject;
+
+import ch.epfl.sweng.erpa.R;
+import ch.epfl.sweng.erpa.model.Game;
+import ch.epfl.sweng.erpa.services.GameService;
+import ch.epfl.sweng.erpa.services.RemoteServicesProvider;
+import ch.epfl.sweng.erpa.services.dummy.DummyRemoteServicesProvider;
+
+public class GameViewerActivity extends Activity {
+
+    public static final String EXTRA_GAME_KEY = "game";
+    @Override
+    @Inject
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_game_viewer);
+
+        //this should get injected
+        RemoteServicesProvider rsp = new DummyRemoteServicesProvider();
```

Comment:
NO, never instanciate an rsp by hand, use `@Inject RemoteServicesProvider rsp` in the class and extend `DependencyConfigurationAgnosticActivity` (Don't forget to add `if (!dependenciesConfigures()) return` after `super.onCreate...`